core/src/main/java/tc/oc/pgm/api/Datastore.java

```diff
@@ -24,6 +25,22 @@
    */
   Settings getSettings(UUID uuid);
 
+  /**
+   * Set the skin for a given player {@link UUID}.
+   *
+   * @param uuid the {@link UUID} of a player
+   * @param skin the {@link Skin} of the player
+   */
+  void setSkin(UUID uuid, Skin skin);
+
+  /**
+   * Get the skin for a given player {@link UUID}.
+   *
+   * @param uuid The {@link UUID} of a player
+   * @return A {@link Skin}
+   */
+  Skin getSkin(UUID uuid);
```

Comment:
Skins will explode the size of the database.. are we sure?

Reply:
Oh, I just realized it's only implemented by cache, ok.

Reply:
Yes, my thought as well :+1: